functionaltests/internal/kbclient/client.go

```diff
@@ -67,14 +76,63 @@ func prepareRequest(c *Client, method, path string, body any) (*http.Request, er
 		return nil, fmt.Errorf("cannot create HTTP request: %w", err)
 	}
 
-	req.Header.Add("Authorization", fmt.Sprintf("ApiKey %s", c.apikey))
 	req.Header.Add("kbn-xsrf", "true")
 	req.Header.Add("Content-Type", "application/json")
 	req.Header.Add("Elastic-Api-Version", c.SupportedAPIVersion)
+	if super {
+		userPass := fmt.Sprintf("%s:%s", c.superUsername, c.superPassword)
+		basicAuth := base64.StdEncoding.EncodeToString([]byte(userPass))
+		req.Header.Add("Authorization", fmt.Sprintf("Basic %s", basicAuth))
+	} else {
+		req.Header.Add("Authorization", fmt.Sprintf("ApiKey %s", c.apiKey))
+	}
 
 	return req, nil
 }
 
+// sendRequest sends a http.Request with the provided method, path and body to Kibana
+// and returns the response body bytes if the status code is 200.
+// It will also handle non-200 status codes accordingly, if the handler is defined.
+func (c *Client) sendRequest(
+	ctx context.Context,
+	method string,
+	path string,
+	body any,
+	super bool,
+	handleRespError func(statusCode int, body []byte) error,
+) ([]byte, error) {
+	req, err := c.prepareRequest(method, path, body, super)
+	if err != nil {
+		return nil, fmt.Errorf("cannot prepare request: %w", err)
+	}
+
+	req = req.WithContext(ctx)
+	resp, err := c.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("cannot perform http request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read response body: %w", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("cannot read response body for  %s %s: %w", method, path, err)
```

A minor suggestion to help locate this error if it happens. We may be reading the body in different places and the body may not contain enough information to pinpoint where the issue is happening.